boards/native/doc/arch_soc.rst

```diff
@@ -304,138 +304,185 @@ side-effects.
+.. figure:: layering_natsim.svg
+    :align: center
+    :alt: Zephyr layering in a native simulator build
+    :figclass: align-center
+
+    Zephyr layering when built against an embedded target (left), and targeting a native_simulator
+    based board (right)
+
+.. figure:: components_natsim.svg
     :align: center
-    :alt: Zephyr layering in native build
+    :alt: native_sim boards and the native simulator
     :figclass: align-center
 
-    Zephyr layering when built against an embedded target (left), and
-    targeting a POSIX arch based board (right)
+    Relationship between Zephyr, the native_sim target and the native simulator
+
+When building targeting Zephyr's :ref:`native_sim<native_sim>` board, we build our embedded SW,
+that is, our application, the Zephyr kernel, and any subsystems and drivers we have selected,
+with the :ref:`POSIX architecture<posix_arch_design_archl>` and the
+:ref:`inf_clock<posix_arch_design_socl>` SOC layers.
+The result of this build is a pre-linked elf library, which contains what we can call the
+embedded SW.
+Then the `native simulator <https://github.com/BabbleSim/native_simulator/>`_ runner will be built.
+And after both the "embedded SW" and the runner will be linked together to form the final Linux
+executable.
+This final executable is typically called ``zephyr.exe`` and can be run or debugged just like any
+other normal Linux executable.
+
+The native simulator runner provides the Linux program entry point, command line argument parsing,
+the HW models scheduler, as well as a component to emulate the CPU start/stop and CPU thread
+switching.
+It also provides a mechanism to register functions which need to be run at different points of the
+executable lifetime.
+When targeting native_sim, the native simulator is also built with some basic HW models like a
+system timer and an interrupt controller.
+You can find more information on these in the
+`native simulator design documentation <https://github.com/BabbleSim/native_simulator/blob/main/docs/Design.md>`_.
+
+The native_sim target is a single microcontroller (MCU) target with simple HW models. Other targets
+like the :ref:`simulated nRF5340 (nrf5340bsim)<nrf5340bsim>` are multi MCU targets. Where one
+embedded Zephyr image can be build for each MCU, and all MCU images and the runner are assembled
+together into one executable with more elaborate HW models of those SOCs.
+
+Native simulator runner context and the embedded context
+========================================================
+
+It is worth noting that the embedded SW library is first pre-linked. That is that all symbols which
+can be resolved inside that library will be resolved. And that, after, all these library symbols,
+except a selected few marked with an special annotation, will be hidden from further linking.
+In this way, the runner link stage will not link to or conflict with any of these hidden symbols,
+neither from the runner itself or from other embedded CPUs libraries.
```

Comment:
what do you mean exactly by "embedded CPU libraries"? The pre-linked lib that has the embedded context, or some other library

Reply:
I mean "The pre-linked lib that has the embedded context"  :)